core/communication_manager.py

```diff
@@ -90,9 +94,14 @@ def _create_session(self, payload):
         # NOTE: We removed the 'optimizer_type' argument since for the MVP we're
         # only considering the 'FederatedAveragingOptimizer' for now.
         # TODO: We need to incorporate session id's when we're ready.
+        if not self.dataset_manager:
+            raise Exception("Communication Manager needs to be configured first!")
```

Comment:
I think this is the same Exception as the Exception raised by Comm Mgr didn't have a Scheduler? Can you differentiate them.